invenio_rdm_records/resources/serializers/datacite/schema.py

```diff
@@ -94,20 +94,36 @@ def get_affiliation(self, obj):
                 }
                 identifiers = affiliation.get("identifiers")
                 if identifiers:
-                    # PIDS-FIXME: DataCite accepts only one, how to decide
-                    identifier = identifiers[0]
+                    # FIXME: Make configurable
+                    DATACITE_AFFILIATION_IDENTIFIER_TYPES_PREFERENCE = (
+                        "ror",
+                        "isni",
+                        "gnd",
+                    )
+                    identifier = get_preferred_identifier(
+                        DATACITE_AFFILIATION_IDENTIFIER_TYPES_PREFERENCE, identifiers
+                    )
+                    if not identifier:
+                        identifier = identifiers[0]
+                        identifier["scheme"] = "Other"
+
                     id_scheme = get_scheme_datacite(
                         identifier["scheme"],
                         "VOCABULARIES_AFFILIATION_SCHEMES",
-                        default=identifier["scheme"],
-                    )
-
-                    if id_scheme:
-                        aff["affiliationIdentifier"] = identifier["identifier"]
-                        aff["affiliationIdentifierScheme"] = id_scheme.upper()
+                        default=identifier["scheme"].upper(),
                         # upper() is fine since this field is free text. It
                         # saves us from having to modify invenio-vocabularies
                         # or do config overrides.
+                    )
+
+                    if id_scheme == "ROR":
```

Comment:
@tmorrell I'm sorry for coming back to this again. As you mentioned DataCite seems to work with ORCID even if this is not a URL, however, does it work if it is a URL? If the answer is yes, then we can simplify the code into:

```python
 if id_scheme:
                         aff["affiliationIdentifier"] = idutils.to_url(identifier_value)
                         aff["affiliationIdentifierScheme"] = id_scheme
```

This would remove the need to tread schemes differently, simplifies code, and makes it more maintainable IMO.

Once again, sorry for the loop hehe

Reply:
That would be an elegant solution if all identifiers were supported in to_url. ISNI is on the RDM metadata scheme, but not in to_url. I might be able to add an ISNI url format to idutils, but it seems a bit beyond the scope of this PR.

Reply:
ahhhh you are right, I had assumed that `to_url` would return the value if the URL was not configured, but it seems it returns `""`. See [here](https://github.com/inveniosoftware/idutils/blob/master/idutils/__init__.py#L1008). 

Then I'd suggest:
1. We merge this to get the fix in
2. We open an issue to re-work it properly, and evaluate the change on default behaviour of idutils

Thanks Tom!